### PR TITLE
Desktop: Resolves/Fixes #4362: Simplify Tray Icons and Fix Ubuntu AppIndicator

### LINKED ISSUE #4362: Ubuntu AppIndicator (AKA Tray Icon)
I'm copying this over from [the Discourse](https://discourse.joplinapp.org/t/ubuntu-appindicator-aka-tray-icon/13904) because I didn't realize that bug reports had moved to GitHub.

I recently discovered that I actually have a third-party Snap installation of Joplin installed on my system. In addition to the basics of Snap packaging, there seems to be one single difference between the Snap version and the official version, namely changes to `joplin.desktop` that allow Joplin to properly display its AppIndicator (AKA "tray icon") on Ubuntu. You can view the changes here:

https://github.com/MrCarroll/joplin-snap/commit/428d39046ba3e49076a6b1fb9a50a8262914f309

This specifically fixes the bug referenced in the Joplin code here:

https://github.com/laurent22/joplin/blob/fca8f71f4a1077feca26fed2b5aff00ad153e360/packages/lib/models/Setting.ts#L669

The link in the code suggests there's a fix:

http://www.webupd8.org/2017/04/fix-appindicator-not-working-for.html

I did some further digging online, and `StartupWMClass=Joplin` and `XDG_CURRENT_DESKTOP=Unity` appear to be two known workarounds for the AppIndicator not working in Ubuntu.

1. https://github.com/pop-os/pop/issues/84
2. https://github.com/ubuntu/gnome-shell-extension-appindicator/issues/74

The main reason I discovered that Joplin's official distribution doesn't support the AppIndicator is that the AppIndicator for the Snap version is full-color, standing out from and kind of clashing with the monochrome AppIndicator icons adjacent to it:

Linux tray

I guess my questions are as follows:

1. Could the AppIndicator fix from the Snap be integrated into the official Joplin distribution?
2. Could the AppIndicator icon on Linux (or at least Ubuntu, idk about other distros) be changed to use the same resource as the Menu Bar icon on macOS?

Mac tray

Note: this function appears to be where it happens:

https://github.com/laurent22/joplin/blob/28f75449d70db1c92c0056459d50abcc713cfda4/packages/app-desktop/ElectronAppWrapper.ts#L267-L279

And,

3. Though this is kind of off-topic, would it feasible or realistic for Joplin to have an official Snap and/or Flatpak/Flathub distribution? (Having both would be ideal because different Linux distros have different preferred "app stores"; Ubuntu, for example only ships with Snap, and Flathub must be added manually.) I see this was requested in #738.

FWIW the Menu Bar icon also seems slightly out of scale with its neighbors on Big Sur, though this problem also affects Nextcloud Desktop:

https://github.com/nextcloud/desktop/issues/2584

(If you figure out the icon scaling issue, I might suggest the same fix over at Nextcloud.)

Thanks!

### PR DESCRIPTION
**Pull request by elsiehupp**
I haven't actually been able to test this since I'm running into build problems at the moment.

### CODE REVIEW

**Comment by elsiehupp** [comment]
I just added editable SVG versions of the icons.

I didn't mention this in the [Issue](https://github.com/laurent22/joplin/issues/4362) or the [Discourse thread](https://discourse.joplinapp.org/t/ubuntu-appindicator-aka-tray-icon/13904), but I also changed the color of the dev-build tray icon to make it easier to distinguish from the Linux icon. For the colors of the dev-build tray icon, I just inverted the hue of the blue gradient while leaving the saturation and luminance as is.

**Comment by elsiehupp** [comment]
@mrcarroll @casasfernando if you want to pipe in...

**Comment by MrCarroll** [comment]
The icons look good to me, I'd say it's an easy win for visual consistency and looks being in line with a majority of other Linux applications, there might be a small shock for people who have become accustomed to the older design but beyond people generally being averse to change I'd imagine most people would happy for the consistency.

**Comment by elsiehupp** [comment]
I know that some Linux desktop environments (I think Xfce?) allow for large/hicolor tray icons, but it's [a known issue that Electron hardcodes the tray icon size at 22px](https://github.com/electron/electron/issues/23287). Basically, if Electron starts allowing Xfce-style tray icons, it may be worth revisiting this.

FWIW it may also be nice to let people choose the tray icon color in the preferences. Nextcloud, for example, lets users choose between monochrome and hicolor icons. Adding this to the preferences UI might be a little more of an undertaking, though.

**Comment by casasfernando** [comment]
I like the proposed icons.
More specifically in the Linux case, being the "J" transparent, I'm not sure how it will look on a solid black panel (the J will become solid black as well in that scenario). In any case I think monochrome tray icons fits better in the panel than color icons.
Having said that, being able to choose to use monochrome or color tray icon like Nextcloud would be ideal. I really love that option in Nextcloud since depending on my panel color (in ElementaryOS it may change according to the desktop background picture) I would use one of another for better contrast and visibility.

Hope this PRs get merged soon. Thanks for your work @elsiehupp !

**Comment by elsiehupp** [comment]
> More specifically in the Linux case, being the "J" transparent, I'm not sure how it will look on a solid black panel (the J will become solid black as well in that scenario).

Here are the screenshots of the Linux tinting from the Discourse thread:

598f6c26f5070d7a152b52b83b5a92f6440a0db1
97b375f5de0b8a7976448467099a61140c80b9a8

The color is actually a very light gray, since that's what Ubuntu seems to do with tintable icons.

**Comment by casasfernando** [comment]
It looks good indeed!

Looking forward to this getting merged for the next release.

**Comment by laurent22** [comment]
Sorry I mentioned before that we only distribute one Linux icon, and other icons should be part of packages, or custom distributions. We won't make any icon change at this point.

**Comment by elsiehupp** [comment]
@laurent22 this isn't the application icon. It's a user interface element. You are doing that thing where you don't actually read before you react.

FWIW, this sort of behavior is why I didn't bother pursuing the Ubuntu AppIndicator before. You make Joplin a toxic environment to work in.

@casasfernando apologies for Laurent. This is actually a continuing pattern for him.

**Comment by elsiehupp** [comment]
@laurent22 the changes in this pull request can't actually be made without completely rebuilding the client from source. Again, you did not bother to actually understand this pull request because you could not be bothered to act in good faith.

**Comment by elsiehupp** [comment]
I'm unsubscribing from this thread. Don't @ me.

**Comment by laurent22** [comment]
I know it's not the main app icon, it's the tray icon, but it's still an icon and we aren't making changes to them as I've told you repeatedly.

My main task on this project is to say "no", that's my answer to probably 90% of queries, because that's the only way to keep things manageable. On the other hand, you clearly don't take "no" for an answer as you keep creating more posts and pull requests related to icons and logos, even though this is very low priority for this project.

You might be disappointed but it's on you really, I think I've been clear that there was no plan to change icons or logos.

By the way I remember about 3-4 times when people said this project was toxic and it's always a situation like this: someone wants something, they don't get what they want, therefore the project is toxic. Thankfully the vast majority of users and contributors to this project are capable of discussing their ideas like adults.

File: Joplin_install_and_update.sh [hunk 1]
```diff
@@ -177,7 +177,7 @@ then
 
     # On some systems this directory doesn't exist by default
     mkdir -p ~/.local/share/applications
-    echo -e "[Desktop Entry]\nEncoding=UTF-8\nName=Joplin\nComment=Joplin for Desktop\nExec=${HOME}/.joplin/Joplin.AppImage\nIcon=joplin\nStartupWMClass=Joplin\nType=Application\nCategories=Office;" >> ~/.local/share/applications/appimagekit-joplin.desktop
+    echo -e "[Desktop Entry]\nEncoding=UTF-8\nName=Joplin\nComment=Joplin for Desktop\nExec=env XDG_CURRENT_DESKTOP=Unity ${HOME}/.joplin/Joplin.AppImage\nIcon=joplin\nStartupWMClass=Joplin\nType=Application\nCategories=Office;" >> ~/.local/share/applications/appimagekit-joplin.desktop
```
Comment (CalebJohn) [inline Joplin_install_and_update.sh:180]: Can you explain this change? Is it necessary? 
Not all Joplin linux users are on Ubuntu, and even those that are aren't necessarily using the Unity desktop.

**Pull request closed without merging**